posts/2024-11-19-24.0.0.12-beta.adoc

```diff
@@ -0,0 +1,234 @@
+seo-description: The 24.0.0.12-beta release enhances inbound SOAP message validation in XML Web Services to simplify message debugging and make your web services and clients more resilient.
+blog_description: The 24.0.0.12-beta release enhances inbound SOAP message validation in XML Web Services to simplify message debugging and make your web services and clients more resilient.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Enhanced message validation for XML Web Services in 24.0.0.12-beta
+David Mueller <https://github.com/dmuelle>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+The 24.0.0.12-beta release enhances inbound SOAP message validation in XML Web Services to simplify message debugging and make your web services and clients more resilient.
+
+
+See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/29770
+// Contact/Reviewer: neuwerk
+// // // // // // // //
+[#xmlws]
+== Fine-tuning XML Web Services inbound SOAP message validation
+
+Open Liberty's XML Web Services features now support fine-grained message validation for inbound SOAP messages. This enhancement provides more control over message validation options.
+In Open Liberty 24.0.0.12-beta, you can configure message validation using new attributes in the `server.xml` file. These attributes are available for the `webService` and `webServiceClient` elements.
+
+
+[options="header"]
+|=======================
+| Attribute | Description
+| `enableSchemaValidation` | Enable full validation against the XML schema
+| `enableDefaultValidation` | Enable or disable default validation for JAXB
+| `ignoreUnexpectedElements` | Use default validation while ignoring `UnmarshallExceptions: Unknown Element` errors
+|=======================
+
+By using these attributes, you can tailor message validation to your specific needs and improve the security and reliability of your SOAP-based web services.
+
+These attributes require the one of the following XML Web Services features to be enabled in your `server.xml` file:
+
+* xref:{url-prefix}/docs/latest/reference/feature/xmlWS-4.0.html[Jakarta XML Web Services
+4.0] (`xmlWS-4.0`)
+* xref:{url-prefix}/docs/latest/reference/feature/xmlWS-3.0.html[Jakarta XML Web Services
+4.0] (`xmlWS-3.0`)
+* xref:{url-prefix}/docs/latest/reference/feature/jaxws-2.2.html[Java Web Services] (`jaxws-2.2`)
+
+You can apply the configuration to web services (`webService`) or web service clients (`webServiceClient`), either globally, or to an individual client or web service implementation.
+
+The examples in the following sections show how to enable each option. To disable any of these options, set the corresponding attributes to `false`.
```

Comment:
> The examples in the following sections show how to enable each option. To disable any of these options, set the corresponding attributes to `false`.

I feel like this is kind of obvious, but OTOH, are we really saying that some of these are enabled by default in some situations, while others are not? So some might be `true` by default, but others `false`, it depends? Feel like we should have more clarity here